### PR TITLE
ParameterAcceptor: check that we set values correctly for multiple live objects of the same class

### LINKED ISSUE #16520: Does add_parameter() set multiple variables when called multiple times?
One of my students, @jerett-cc, is working with @tamiko 's Ryujin code. In that code, the main class's constructor is calling `add_parameter()` with its member variables as argument. When the input file is parsed, these member variables are set via the associated "action".

Now, in his variation of the Ryujin code, he needs to create multiple instances of the main object (because he is creating some multilevel hierarchy). What then happens is that every one of these objects, in their constructor calls, call `add_parameter()` with the same name, but different member variables (i.e., the same member variable but belonging to different objects). The question is: If you do this, when the input file is parsed, does that set *all* of the variables, or only one (e.g., the one where `add_parameter()` was called with first, or perhaps last)?

This issue is just a reminder for me to verify that it really does initialize *all* member variables. (I think it does, but I want to write a small test that checks that.)

### PR DESCRIPTION
**Pull request by tamiko**
Closes #16520

### CODE REVIEW

File: tests/parameter_handler/parameter_acceptor_12.output [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+DEAL::1 - 1 - 1
+DEAL::2 - 2 - 2
```
Comment (tamiko) [inline tests/parameter_handler/parameter_acceptor_12.output:3]: Yes, it should work and it does indeed work as expected. :-)

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Well, that was quick!

@jerett-cc FYI.

**Comment by jerett-cc** [comment]
Thanks for the FYI. We’ll need to discuss the other issue separately. I think I see that that will not work in the way that we thought.-Jerett CherryOn Jan 22, 2024, at 3:36 PM, Wolfgang Bangerth ***@***.***> wrote:﻿
Well, that was quick!
@jerett-cc FYI.

**Review by masterleinad — APPROVED** [review]

**Pull request merged**